src/views/Task/Update.js

```diff
@@ -1,29 +1,178 @@
-import React from 'react'
+import React, { Component } from 'react'
 import Layout from '../../components/Layout'
+import task from '../../api/task'
+import DatePicker from '../../components/DatePicker'
+import moment from 'moment'
+import Alert from '../../components/Alert'
 
-export default () => (
-  <Layout>
-    <h2>Edit Task</h2>
-    <label>
-      Title
-      <input type="text" />
-    </label>
-    <label>
-      Start Date
-      <input type="date" />
-    </label>
-    <label>
-      Deadline
-      <input type="date" />
-    </label>
-    <label>
-      Description
-      <textarea />
-    </label>
-    <label>
-      Urgency
-      <input type="text" />
-    </label>
-    <input type="submit" value="Save Changes" />
-  </Layout>
-)
+export default class Update extends Component {
+  state = {
+    startDate: moment(),
+    endDate: moment(),
+    title: "",
+    description: "",
+    urgency: "",
+    error: null
+  }
+
+  /**
+   * Gets Task arguments and Setstate
+   */
+  getTask = async () => {
+    const task = await task.get(this.props.match.params.id);
+    this.setState({
+      title: task.title,
+      description: task.description,
+      urgency: task.urgency,
+      startDate: task.startDate,
+      endDate: task.endDate,
+    })
+  }
+
+  componentDidMount = () => {
+    this.getTask()
+  }
+
+  fields = [
+    //["label", "Type", Validation method]
+    ["Title", "text", v => v.length > 0, false],
+    ["Description", "text", v => v.length > 0, true],
+    ["Urgency", "text", v => v.length > 0, false]
+  ]
+
+  /**
+ * Sets the selected  date
+ * @param {Date} date
+ */
+  updateStartDate = (date) => {
+    this.setState({ startDate: date })
+    this.setState({ endDate: date })
+  }
+
+  /**
+   * Sets selected date of deadline
+   * @param {Date} date
+   */
+  updateEndDate = (date) => {
+    this.setState({ endDate: date })
+
```

Comment:
an empty line

Reply:
line 58

Reply:
Not all empty lines should be removed, it's ok if it makes the code easier to reason about. Often if is used to separate sections of code into logical pieces.